diff -Pru plugins/com.napolitano.cordova.plugin.intent.orig/src/android/IntentPlugin.java plugins/com.napolitano.cordova.plugin.intent/src/android/IntentPlugin.java
--- plugins/com.napolitano.cordova.plugin.intent.orig/src/android/IntentPlugin.java	2018-03-25 19:27:15.319070288 -0700
+++ plugins/com.napolitano.cordova.plugin.intent/src/android/IntentPlugin.java	2018-03-25 19:28:31.985653007 -0700
@@ -11,6 +11,7 @@
 import android.content.ClipData;
 import android.content.Intent;
 import android.os.Build;
+import android.os.Bundle;
 import android.util.Log;
 
 import org.apache.cordova.CallbackContext;
@@ -142,14 +143,21 @@
         try {
             intentJSON = new JSONObject();
 
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
-                if(items != null) {
-                    intentJSON.put("clipItems", new JSONArray(items));
-                }
+            if(items != null) {
+                intentJSON.put("clipItems", new JSONArray(Arrays.asList(items)));
             }
 
             intentJSON.put("type", intent.getType());
-            intentJSON.put("extras", intent.getExtras());
+            Bundle extraBundle = intent.getExtras();
+            if (extraBundle != null) {
+                JSONObject extras = new JSONObject();
+                for (String k : extraBundle.keySet()) {
+                    Object o = extraBundle.get(k);
+                    extras.put(k, o);
+                }
+                intentJSON.put("extras", extras);
+            }
+            
             intentJSON.put("action", intent.getAction());
             intentJSON.put("categories", intent.getCategories());
             intentJSON.put("flags", intent.getFlags());
@@ -167,4 +175,4 @@
         }
     }
 
-}
\ No newline at end of file
+}
